assignments/assignment02.rb

```diff
@@ -0,0 +1,187 @@
+# ========================================================================================
+#  Assignment submitted by Gordon Turibamwe
+#
+# ========================================================================================
+# Assignment 2
+# ========================================================================================
+
+# ========================================================================================
+#  Problem 1 - `to_sentence`
+
+# implement method `to_sentence`
+
+# creates an english string from array
+
+def to_sentence(ary)  
+  # PSEUDO CODE
+  # get array
+  # create new string
+  # add each word to new string
+  # if more than 2 words, add "and" before the last word
+  # print sentence
+
+  sentence = ""
+  ary.each do |word|
+    if ary.length > 2 && ary.index(word) == ary.length - 1
+      sentence += "and "
+    end
+    sentence += word.to_s + ',' + " "
+  end
+  sentence = sentence.chomp(", ")
+print sentence
+return sentence #added this tests
+end
+
+# Your method should generate the following results:
+#to_sentence []                       #=> ""
+#to_sentence ["john"]                 #=> "john"
+#to_sentence ["john", "paul"]         #=> "john and paul"
+#to_sentence [1, "paul", 3, "ringo"]  #=> "1, paul, 3 and ringo"
+
+# ========================================================================================
+#  Problem 2 - `mean, median`
+
+# implement methods "mean", "median" on Array of numbers
+def mean(ary)
+  # PSEUDO CODE
+  # Inititate sum variable
+  # iterate arry adding each velue to sum variable
+  # divide sum variabe by length of the array and convert to a float
+
+  sum = 0.0
+  mean = 0.0
+  ary.each do |num|
+    sum += num
+    mean = sum / ary.length
+  end
+  print mean.round(1)
+  return mean.round(1)
+end
+
+def median(ary)
+  # PSEUDO CODE
+  # if the length is odd
+  # get the ary length and 1 to it divide by 2 to get the index of the value
+  # if the length is even
+  # get the ary length add 1 and find the index nums between the tow values and get their average
+
+  length = ary.length
+  median = 0.0
+  if length % 2 > 0
+    median = ary[length / 2.0]
```

Comment:
you know the length is even, so you don't have to divide by 2.0, you can just divide by 2